### PR TITLE
44802020 - Facebook GCN

### PR DESCRIPTION
**Pull request by johnvparsons99**
This PR is to merge my attempt at  Task 2: Graph Convolutional Neural Net Model for the Facebook Large Page-Page Network Dataset. The model aims to be able to categorize the pages into 1 of the 4 categories. It does so with 72% validated accuracy.

### CODE REVIEW

**Comment by gayanku** [comment]
- PR on Nov 1
- well detailed commit messages and PR
- Readme is well done with background, results, diagrams. 
- multi layer (3) layer GCN, Low accuracy (+70%)
- **Valid for EASY ( subject to Fully supervised and low accuracy due to using two Adj matrices)**

- Has used fully supervised split (80/20). In future make sure that n_train < n_test for Semi supervised. 
- maybe use GCNLayer rather than FaceGCNLayer. There is no specialized behavior related to the Face dataset in the GCN layer. It can be re-used in other datasets as well unmodified. Also avoid naming such as makeMyModel; you can be more descriptive. 
- Avoid comments that are unnecessary/redundant as they do not add any new information ( ex: My custom network layer). 
- Put all methods (preprocess, TSNE etc) away from the driver file. This promotes reuse of the code. Driver should only put all the modules together.
- Low accuract due to the fact of using 2 adj matrices ( one for test another for train). Use a single Adj.  See work from kipf on gitgub

**Comment by shakes76** [comment]
## TF/Torch Usage
Adequate use and implementation
## Good Practice (Design/Commenting)
Good spacing and comments
No Header blocks -1
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present 
No Data leakage 
Difficulty: Normal -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No feedback required
Request Description could be more informative -1
